app/Models/User.php

```diff
@@ -40,6 +49,56 @@ class User extends Authenticatable
      * @var array<string, string>
      */
     protected $casts = [
-        'email_verified_at' => 'datetime',
+        'email_verified_at' => 'date:d-m-Y',
+        'birthday' => 'date:d-m-Y',
     ];
+
+    public function posts(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Post::class, 'user_id', 'id');
+    }
+
+    //Lấy ra list người đã follow user đó
+    public function follows(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Follow::class, 'follower_id', 'id');
+    }
+
+    //Lấy ra list người user đó đã follow (thông qua user_id)
+    public function had_follow(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Follow::class, 'user_id', 'id');
+    }
+
+    public function role(): \Illuminate\Database\Eloquent\Relations\HasOne
+    {
+        return $this->hasOne(Role::class, 'id', 'role_id');
+    }
+
+    public function status(): \Illuminate\Database\Eloquent\Relations\HasOne
+    {
+        return $this->hasOne(Status::class, 'id', 'status_id');
+    }
+
+    public function reviews(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Review::class, 'user_id', 'id');
+    }
+
+    public function replies(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Reply::class, 'user_id', 'id');
+    }
+
+    // bị report
+    public function reports(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
+        return $this->hasMany(Report::class, 'report_id', 'id');
+    }
+
+    // đi report
+    public function user_report(): \Illuminate\Database\Eloquent\Relations\HasMany
+    {
```

Comment:
```suggestion
    // reports created by user
    public function userReports(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
```
camelCase, comment english